skimage/exposure/tests/test_exposure.py

```diff
@@ -21,9 +21,10 @@
 # Test integer histograms
 # =======================
 
-def test_wrong_source_range():
-    im = np.array([-1, 100], dtype=np.int8)
-    with pytest.raises(ValueError):
+@pytest.mark.parametrize('dtype', [np.int8, np.float32])
+def test_wrong_source_range(dtype):
+    im = np.array([-1, 100], dtype=dtype)
+    with pytest.raises(ValueError, match="Incorrect value for `source_range` argument"):
```

Comment:
The ValueError for each dtype is actually raised from two different locations. So I took the opportunity to make the text identical and test for it to avoid similar surprises in the future. 

I also found two additional cases were the `raise` statement was missing and added small tests.